redash/query_runner/azure_kusto.py

```diff
@@ -91,18 +119,38 @@ def name(cls):
         return "Azure Data Explorer (Kusto)"
 
     def run_query(self, query, user):
-        kcsb = KustoConnectionStringBuilder.with_aad_application_key_authentication(
-            connection_string=self.configuration["cluster"],
-            aad_app_id=self.configuration["azure_ad_client_id"],
-            app_key=self.configuration["azure_ad_client_secret"],
-            authority_id=self.configuration["azure_ad_tenant_id"],
-        )
+        cluster = self.configuration["cluster"]
+        msi = self.configuration.get("msi", False)
+        # Managed Service Identity(MSI)
+        if msi:
+            kcsb = KustoConnectionStringBuilder.with_aad_managed_service_identity_authentication(cluster)
+        # Specific user assigned Managed Service Identity
+        elif msi and self.configuration["user_msi"]:
```

Comment:
I think this won't be called because if `msi` is `True` then it will always stop with the first `if`. You need to have this come before the plain `if msi:`.